### PR TITLE
you can now cmag energy weapons [NOT READY]

### PR DESCRIPTION
**Pull request by spacegeese**
## What Does This PR Do

allows you to use a cmag on energy weapons to shoot harmless rainbow projectiles at people that make FUNNY SOUNDS!!!

pr is NOT ready see to do list i will finish when free time
## Why It's Good For The Game

i was originally gonna make it shoot a hologram flag but sprite box too small, also funny thing good
## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/136994596/0554b1f2-9e3a-4e47-8e51-a8d29c8d901f

## Testing

see above, i did not test hitscan fuck
## Changelog
:cl:
add: you can now cmag all energy weapons
/:cl:

## To do

- [ ] fix weird runtime, image here 
image

- [ ] fix first shot being normal laser
- [ ] make vfx better

### CODE REVIEW

**Comment by spacegeese** [comment]
> Your PR has been put on hold as your GitHub account is not linked to a Paradise forums account.
> 
> Please link your account at https://www.paradisestation.org/forum/custom/link_github_account.php
> 
> _Headcoders may ignore this in specific circumstances (Minor fix PR, trusted contributor from another codebase, anything at their discretion)_

my past 2 prs didn't HAVE this at all

**Comment by TravisAngeI** [comment]
Will the gun be able to be cleaned with soap to re-energy gun it?

File: code/modules/projectiles/guns/energy_guns.dm [hunk 1]
```diff
@@ -252,3 +252,17 @@
 		update_icon()
 	else
 		charge_tick = 0
+
+/obj/item/gun/energy/cmag_act(mob/user)
+	if(!HAS_TRAIT(src, TRAIT_CMAGGED))
+		to_chat(user, "<span class='warning'>Yellow ooze seeps into the [src]'s barrel and trigger...</span class='warning'>")
```
Comment (lewcc) [inline code/modules/projectiles/guns/energy_guns.dm:258]:
```suggestion
		to_chat(user, "<span class='warning'>Yellow ooze seeps into [src]'s barrel and trigger...</span>")
```

File: code/modules/projectiles/guns/energy_guns.dm [hunk 2]
```diff
@@ -252,3 +252,17 @@
...
+		ADD_TRAIT(src, TRAIT_CMAGGED, CLOWN_EMAG)
+		if(HAS_TRAIT(src, TRAIT_CMAGGED))
```
Comment (lewcc) [inline code/modules/projectiles/guns/energy_guns.dm:260]: You just added it here, you shouldn't need to check this again

**Comment by Sirryan2002** [comment]
No Progress on this PR in over a month, when you're ready to continue progress on this PR you are welcome to reopen it.

**Pull request closed without merging**